libafl/src/mutators/tuneable.rs

```diff
@@ -200,17 +199,32 @@ where
             return ret;
         }
 
-        // We will sample using the mutation probabilities.
-        // Doing this outside of the original if branch to make the borrow checker happy.
-        #[allow(clippy::cast_precision_loss)]
-        let coin = state.rand_mut().next() as f32 / u64::MAX as f32;
-        let metadata = TuneableScheduledMutatorMetadata::get_mut(state).unwrap();
-        metadata
-            .mutation_probabilities_cumulative
-            .iter()
-            .position(|i| *i >= coin)
-            .unwrap()
-            .into()
+        if !metadata.mutation_probabilities_cumulative.is_empty() {
+            // We will sample using the mutation probabilities.
+            // Doing this outside of the original if branch to make the borrow checker happy.
+            #[allow(clippy::cast_precision_loss)]
+            let coin = state.rand_mut().next() as f32 / u64::MAX as f32;
+            debug_assert!(coin <= 1.0_f32);
+
+            let metadata = TuneableScheduledMutatorMetadata::get_mut(state).unwrap();
+            debug_assert_eq!(
+                self.mutations.len(),
+                metadata.mutation_probabilities_cumulative.len(),
+                "TuneableScheduler: mutation probabilities do not match with number of mutations"
+            );
+
+            let mutation_id = metadata
+                .mutation_probabilities_cumulative
+                .iter()
+                .position(|i| *i >= coin)
+                .unwrap()
```

Comment:
An alternative fix could have been to change this to `.ok_or(probs.len() - 1)` (default to the last one), but that would group all possible errors into the same one.